ports/samd/machine_uart.c

```diff
@@ -479,12 +486,18 @@ static void mp_machine_uart_init_helper(machine_uart_obj_t *self, size_t n_args,
 }
 
 static mp_obj_t mp_machine_uart_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 1, MP_OBJ_FUN_ARGS_MAX, true);
+    mp_arg_check_num(n_args, n_kw, 0, MP_OBJ_FUN_ARGS_MAX, true);
 
     // Get UART bus.
-    int uart_id = mp_obj_get_int(args[0]);
+    int uart_id = MICROPY_HW_DEFAULT_UART_ID;
+    int arg_offset = 0;
+
+    if (n_args > 0 && mp_obj_get_int(args[0]) <= SERCOM_INST_NUM) {
```

Comment:
No need for the second check here against `SERCOM_INST_NUM`, that will be handled just below.

(Comment also applied to other similar code in I2C/SPI.)

Reply:
Done. Changed for UART/SPI. I2C did not need the change.